s2spy/calendar_shifter.py

```diff
@@ -0,0 +1,169 @@
+import copy
+from typing import Dict
+from typing import List
+from typing import Union
+import xarray as xr
+import s2spy.time
+
+
+def _gap_shift(
+    interval: s2spy.time.Interval, shift: Union[str, Dict[str, int]]
+) -> Dict[str, int]:
+    """
+    Shift a gap from a calendar interval by an input of a pandas-like
+    frequency string (e.g. "10d", "2W", or "3M"), or a pandas.DateOffset
+    compatible dictionary such as {days=10}, {weeks=2}, or {months=1, weeks=2}..
+    Args:
+        gap: a pandas DateOffset from a calendar instance.
+        shift: the shift for the gap
+    Returns:
+        dict with gap offset by shift
+    """
+    # pylint: disable=protected-access
```

Comment:
If we want to use this utility function outside of the interval class, we should put it elsewhere and not have it be a protected member. Let's move `_parse_timestring` to utils.py, and name it something like `parse_freqstr_to_dateoffset`.

Otherwise _parse_timestring's functionality might change and break the behavior that the calendar shifter expects.

Reply:
Good idea! I will move it to utils.